src/main/java/com/feedzai/commons/sql/abstraction/batch/AbstractBatch.java

```diff
@@ -365,6 +365,17 @@ protected AbstractBatch(final DatabaseEngine de, final int batchSize, final long
         this(de, null, batchSize, batchTimeout, maxAwaitTimeShutdown);
     }
 
+    /**
+     * A functional interface to represent a {@link java.util.function.BiConsumer} that throws an exception.
+     *
+     * @param <T> the type of the first argument to the operation.
+     * @param <R> the type of the second argument to the operation.
+     */
+    @FunctionalInterface
+    public interface ThrowingBiConsumer<T, R> {
+        void accept(T t, R r) throws Exception;
+    }
```

Comment:
If this is a public interface (even more a so generic one, both in the arguments as in the exception handling), then I would move this to a separate file.

If this is intended to be used just by this class, then I would 
- make it `private`,
- rename the method to something more use-case oriented, eg `FlushConsumer`,
- throw `DatabaseEngineException` instead